### PR TITLE
Add preliminary support for the resume API

### PR DESCRIPTION
**Pull request by orpiske**
**Release Note**
```release-note
NONE
```

### CODE REVIEW

**Comment by orpiske** [comment]
@lburgazzoli FYI, this is to get us started and we can build from here. wdyt?

File: pkg/apis/camel/v1/common_types.go [hunk 1]
```diff
@@ -304,6 +304,8 @@ const (
 	CapabilityTracing = "tracing"
 	// CapabilityMaster defines the master capability
 	CapabilityMaster = "master"
+	// CapabilityResume defines the resume capability
+	CapabilityResume = "resume-kafka"
```
Comment (lburgazzoli) [inline pkg/apis/camel/v1/common_types.go:308]: I guess this need to be scoped as we did for the runtime i.e. CapabilityResumeKafka

**Review by squakez — COMMENTED** [review]
A bit of usage documentation either in the same addon or externally in the official doc would be really appreciated. Without that, the user has no clue of what this is about...

File: pkg/apis/camel/v1/common_types.go (hunk 1 above)
Comment (orpiske) [reply pkg/apis/camel/v1/common_types.go:308]: +1. I'll adjust. Thanks.

**Comment by orpiske** [comment]
> A bit of usage documentation either in the same addon or externally in the official doc would be really appreciated. Without that, the user has no clue of what this is about...

👍 

We have some documentation here: https://camel.apache.org/components/next/eips/resume-strategies.html. I think it needs some documentation about the trait + a link to that page.

**Comment by squakez** [comment]
> +1
> 
> We have some documentation here: https://camel.apache.org/components/next/eips/resume-strategies.html. I think it needs some documentation about the trait + a link to that page.

That would be perfect, thanks!

**Comment by orpiske** [comment]
BTW, I am also working on some examples, similar to the ones I have for Camel Core. I plan to submit them to the examples repo once this is merged in and stabilized.

**Review by squakez — APPROVED** [review]

**Comment by orpiske** [comment]
Hey @lburgazzoli, any other comments about this code? I'd like to merge it today if possible / good enough ...

**Review by lburgazzoli — APPROVED** [review]

**Comment by orpiske** [comment]
Thanks for the review @lburgazzoli and @squakez. I am going to merge this now.

**Comment by tadayosi** [comment]
BTW, we need to add its entry to doc gen script and run `make generate` so that the addon doc is generated. Not sure if it's right time to do so though, as we are in the process of releasing 1.10.0.
https://github.com/apache/camel-k/blob/main/script/gen_doc.sh#L30

**Comment by squakez** [comment]
True, we can easily create a new PR with the generated doc. There are still some issue to polish on camel k runtime, so, we still have time to do so.

**Pull request merged**